mne/time_frequency/tfr.py

```diff
@@ -2024,8 +2036,94 @@ def __abs__(self):
         """Take the absolute value."""
         return EpochsTFR(info=self.info.copy(), data=np.abs(self.data),
                          times=self.times.copy(), freqs=self.freqs.copy(),
+                         events=self.events.copy(), event_id=self.event_id.copy(),
                          method=self.method, comment=self.comment)
 
+    def __len__(self):
+        """Get the number of epochs."""
+        return len(self.events)
+
+    def _keys_to_idx(self, keys):
+        """Find entries in event dict."""
+        keys = [keys] if not isinstance(keys, (list, tuple)) else keys
+        try:
+            # Assume it's a condition
+            return np.where(np.any(
+                np.array([self.events[:, 2] == self.event_id[k]
+                          for k in _hid_match(self.event_id, keys)]),
+                axis=0))[0]
+        except KeyError as err:
+            raise KeyError(msg)
+
+    def __getitem__(self, item):
+        """Return an EpochsTFR object with a copied subset of epochs.
+
+        Parameters
+        ----------
+        item : slice, array-like, str, or list
+
+        Returns
+        -------
+        epochs : instance of Epochs
+
+        Notes
+        -----
+        See BaseEpochs.__getitem__ for use cases.
+        The use of metadata in EpochsTFR is not yet functional.
+        """
+        return self._getitem(item)
+
+    def _getitem(self, item, copy=True, drop_event_id=True,
+                 return_indices=False):
```

Comment:
can you factorize this function with the Epochs class eg with a Mixin object from which both Epochs and EpochsTFR would inheritate?